src/codewhisperer/util/authUtil.ts

```diff
@@ -72,7 +72,10 @@ export class AuthUtil {
                 vscode.commands.executeCommand('aws.codeWhisperer.refreshStatusBar'),
                 vscode.commands.executeCommand('aws.codeWhisperer.updateReferenceLog'),
             ])
-            await vscode.commands.executeCommand('aws.codeWhisperer.enableCodeSuggestions')
+            // To check valid connection
+            if (this.isValidEnterpriseSsoInUse() || this.isBuilderIdInUse()) {
```

Comment:
isBuilderIdInUse doesn't check connection expired. Does this work for the case when user is using Builder ID and expired(should not show)?

Reply:
I think logic covers all the basics

` if (this.isValidEnterpriseSsoInUse() || (this.isBuilderIdInUse()&& !this.isConnectionExpired())) `

Reply:
yea, I mean the current code only has `this.isValidEnterpriseSsoInUse() || this.isBuilderIdInUse()`

Reply:
Yes Updating  accordingly